src/include/ASTree.h

```diff
@@ -0,0 +1,104 @@
+#pragma once
+
+#include <Token.h>
+#include <vector>
+#include <ios>
+#include <sstream>
+
+struct Statement { };
+
+//////////////////////////////////////////
+/// Expressions
+
+struct Expression : public Statement { };
+
+class ExpressionGroup : public Expression {
```

Comment:
I think this should all be `struct`s rather than `class`. I know they *technically* have a constructor method but they're basically just "dumb bags of data" (plus the parent `Statement` is a `struct`).

Reply:
thats a good point! wasn't sure if we would need functions on them, but you're right they'll probably just hold stuff